modules/ppcp-api-client/src/Repository/CustomerRepository.php

```diff
@@ -56,4 +58,17 @@ public function customer_id_for_user( int $user_id ): string {
 
 		return $this->prefix . (string) $user_id;
 	}
+
+	/**
+	 * Generates a unique id based on the length of the prefix.
+	 *
+	 * @return string
+	 */
+	protected function generate_unique_id(): string {
+		$offset = self::CLIENT_ID_MAX_LENGTH - strlen( $this->prefix );
+
+		return strlen( uniqid() ) > $offset
+			? $this->prefix . substr( uniqid(), $offset )
+			: $this->prefix . uniqid();
+	}
```

Comment:
This seems difficult to understand and not sure if correct. Also two `uniqid` calls, possibly returning different lengths.

Maybe better

    return substr( $$this->prefix . strrev( uniqid() ), 0, self::CLIENT_ID_MAX_LENGTH );